xbmc/guilib/imagefactory.cpp

```diff
@@ -46,16 +46,15 @@ IImage* ImageFactory::CreateLoader(const CURL& url)
 
 IImage* ImageFactory::CreateLoaderFromMimeType(const std::string& strMimeType)
 {
-  VECADDONS codecs;
-  ADDON::CBinaryAddonCache &addonCache = CServiceBroker::GetBinaryAddonCache();
-  addonCache.GetAddons(codecs, ADDON::ADDON_IMAGEDECODER);
-  for (auto& codec : codecs)
```

Comment:
Yes, this replace complete the binary addon cache, mostly is only the info like here for supported types needed.

This parts are continuous stored in binary addon manager.

In case now a running addon need to be asked becomes them during active time also stored and can be used by new way.

From performance it is equal with the other.